### DISCUSSION TITLE
Why appwrite is written in php?

### DISCUSSION THREAD

**Discussion by NeroBlackstone** [appwrite/appwrite#1428] [Q&A] [answered]
Could the Appwrite development team share the reason for choosing PHP? I'm curious.Thanks!

**Comment by eldadfux** [comment, marked as answer]
Our team is composed from PHP veterans, and with recent PHP improvments introduced in PHP7 and PHP8 we decide to stick with. PHP gives us a good balance of simplicity, ease-of-use and probably best performance out there for a dynamic, non-compiled language.

Since Appwrite 0.8, Appwrite is using Swoole as an HTTP and Co-Routines framework. Swoole has helped us boost the server performance by over 90%. With Swoole Appwrite has first class async and non-blocking support.

**Reply by NeroBlackstone** [reply]
Thanks for your reply!

**Reply by koodeau** [reply]
Oh, I’m reading this and wondering how many people struggle with async in Rust nowadays, by the way Go could be a nice option if it fits

**Reply by donywan** [reply]
Have you a  plan change php to nestjs or others?

**Reply by Hugyn** [reply]
weird feeling of stop using it after hearing this
